src/Framework/Framework/Controls/HtmlGenericControl.cs

```diff
@@ -344,9 +346,18 @@ private void AddCssStylesToRender(IHtmlWriter writer)
 
         private void AddHtmlAttribute(IHtmlWriter writer, string name, object? value)
         {
-            if (value is string || value == null)
+            if (value is null)
+                writer.AddAttribute(name, null, append: true);
+            if (value is string str)
             {
-                writer.AddAttribute(name, (string?)value, true);
+                writer.AddAttribute(name, str, append: true);
+            }
+            else if (value is AttributeList list)
+            {
+                for (var i = list; i is not null; i = i.Next)
+                {
+                    AddHtmlAttribute(writer, name, i.Value);
+                }
             }
             else if (value is IEnumerable<string>)
```

Comment:
Remove this branch - the implementation is wrong and we should not support this.